src/CADL.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -426,6 +427,22 @@ export function getInputType(
 
             models.set(name, model);
 
+            if(discriminator) {
+                const discriminatorProp = {
+                    Name: discriminator.propertyName,
+                    SerializedName: discriminator.propertyName,
+                    Description: "",
+                    Type: {
+                        Name: "String",
+                        Kind: InputTypeKind.String,
+                        IsNullable: false
+                    } as InputPrimitiveType,
+                    IsRequired: true,
+                    IsReadOnly: false,
```

Comment:
Why readonly is false?  Can we overwrite the value of a discriminator?

Reply:
We need to provide value for discriminator parameter when we call the api